sqlx-macros/src/query_macros/args.rs

```diff
@@ -31,7 +32,19 @@ pub fn quote_args<DB: DatabaseExt>(
                                 .unwrap(),
                         )
                     })
-                    .ok_or_else(|| format!("unknown type param ID: {}", type_).into())
+                    .ok_or_else(|| {
+                        if let Some(feature_gate) = <DB as DatabaseExt>::get_feature_gate(&type_) {
+                            format!(
+                                "optional feature `{}` required for type {} of param #{}",
+                                feature_gate,
+                                type_,
+                                i + 1,
+                            )
+                            .into()
```

Comment:
Nice! This will probably help a lot of folks. Easy to forget to add feature flags.